src/network/tarcap/new_taraxa_capability.cpp

```diff
@@ -0,0 +1,427 @@
+  packets_handlers_->registerHandler(PriorityQueuePacketType::PQ_NewBlockHashPacket, dag_handler);
+  packets_handlers_->registerHandler(PriorityQueuePacketType::PQ_GetNewBlockPacket, dag_handler);
+
+  packets_handlers_->registerHandler(
+      PriorityQueuePacketType::PQ_TransactionPacket,
+      std::make_shared<TransactionPacketHandler>(peers_state_, packets_stats, trx_mgr, dag_blk_mgr, test_state_,
+                                                 conf.network_transaction_interval, node_addr));
+
+  // Non critical packets with low processing priority
+  packets_handlers_->registerHandler(PriorityQueuePacketType::PQ_TestPacket,
+                                     std::make_shared<TestPacketHandler>(peers_state_, packets_stats, node_addr));
+  packets_handlers_->registerHandler(
+      PriorityQueuePacketType::PQ_StatusPacket,
+      std::make_shared<StatusPacketHandler>(peers_state_, packets_stats, syncing_state_, syncing_handler_, pbft_chain,
+                                            dag_mgr, next_votes_mgr, pbft_mgr, conf.network_id, node_addr));
+  packets_handlers_->registerHandler(PriorityQueuePacketType::PQ_GetBlocksPacket,
+                                     std::make_shared<GetBlocksPacketsHandler>(peers_state_, packets_stats, trx_mgr,
+                                                                               dag_mgr, dag_blk_mgr, db, node_addr));
+
+  packets_handlers_->registerHandler(PriorityQueuePacketType::PQ_BlocksPacket,
+                                     std::make_shared<BlocksPacketHandler>(peers_state_, packets_stats, syncing_state_,
+                                                                           syncing_handler_, dag_blk_mgr, node_addr));
+
+  // TODO there is additional logic, that should be moved outside process function
+  packets_handlers_->registerHandler(
+      PriorityQueuePacketType::PQ_GetPbftBlockPacket,
+      std::make_shared<GetPbftBlockPacketHandler>(peers_state_, packets_stats, syncing_state_, pbft_chain, db,
+                                                  conf.network_sync_level_size, node_addr));
+
+  const auto pbft_handler = std::make_shared<PbftBlockPacketHandler>(
+      peers_state_, packets_stats, syncing_state_, syncing_handler_, pbft_chain, pbft_mgr, dag_blk_mgr,
+      conf.network_sync_level_size, node_addr);
+  packets_handlers_->registerHandler(PriorityQueuePacketType::PQ_PbftBlockPacket, pbft_handler);
+  packets_handlers_->registerHandler(PriorityQueuePacketType::PQ_SyncedPacket, pbft_handler);
+
+  thread_pool_.setPacketsHandlers(packets_handlers_);
+}
+
+std::string TaraxaCapability::name() const { return TARAXA_CAPABILITY_NAME; }
+
+unsigned TaraxaCapability::version() const { return TARAXA_NET_VERSION; }
+
+unsigned TaraxaCapability::messageCount() const { return SubprotocolPacketType::PacketCount; }
+
+void TaraxaCapability::onConnect(weak_ptr<dev::p2p::Session> session, u256 const &) {
+  const auto node_id = session.lock()->id();
```

Comment:
this is probably copy&paste, but you are doing lock() 2x and first time do not check if it is ok